tests/test_leanplum_export.py

```diff
@@ -0,0 +1,269 @@
+            assert not os.path.isdir("outputsessions")
+
+    @responses.activate
+    def test_save_files_multiple_uris(self, exporter):
+        n_files = 5
+        base_uri = "https://leanplum_export.storage.googleapis.com/export-5094741967896576-60c43e66-30fe-4e21-9bbd-563d2749b96f"
+        file_types = ["outputsessions", "outputexperiments"]
+        file_uris = [f'{base_uri}-{ftype}-{i}' for ftype in file_types for i in range(n_files)]
+        bucket = 'abucket'
+        prefix = ''
+        file_body = b"data"
+        date = "20190101"
+
+        responses.add(
+             responses.GET,
+             re.compile(r"https://leanplum_export.storage.googleapis.com.*"),
+             body=file_body,
+             status=200)
+
+        with patch('leanplum_data_export.export.storage', spec=True) as MockStorage:
+            self.file_contents = None
+            def set_contents(filename):
+                with open(filename, "rb") as f:
+                    self.file_contents = f.read()
+
+            mock_bucket, mock_client, mock_blob = Mock(), Mock(), Mock()
+            MockStorage.Client.return_value = mock_client
+            mock_client.get_bucket.return_value = mock_bucket
+            mock_bucket.blob.return_value = mock_blob
+            mock_blob.upload_from_filename.side_effect = set_contents
+
+            tables = exporter.save_files(file_uris, bucket, prefix, date, "json")
+            mock_client.get_bucket.assert_called_with(bucket)
+            assert tables == set(file_types)
+
+            for ftype in file_types:
+                for i in range(n_files):
+                    suffix = f"{ftype}/{i}.json"
+                    mock_bucket.blob.assert_any_call(f"{date}/{suffix}")
+                    mock_blob.upload_from_filename.assert_any_call(suffix)
+                    assert self.file_contents == file_body
+                    assert not os.path.isfile(suffix)
+                    assert not os.path.isdir(ftype)
+
+    @responses.activate
+    def test_save_files_improper_file_format(self, exporter):
+        file_uris = ['https://leanplum_export.storage.googleapis.com/export-5094741967896576-60c43e66-30fe-4e21-9bbd-563d2749b96f-outputsessions']
+        bucket = 'abucket'
+        prefix = ''
+        file_body = b"data"
+        date = "20190101"
+
+        responses.add(
+             responses.GET,
+             re.compile(r"https://leanplum_export.storage.googleapis.com.*"),
+             body=file_body,
+             status=200)
+
+        with patch('leanplum_data_export.export.storage', spec=True) as MockStorage:
+            with pytest.raises(Exception) as e:
+                exporter.save_files(file_uris, bucket, prefix, date, "json")
+
+    @responses.activate
+    def test_export(self, exporter):
+        date = "20190101"
+        job_id = "testjobid"
+        file_uris = ['https://leanplum_export.storage.googleapis.com/export-5094741967896576-60c43e66-30fe-4e21-9bbd-563d2749b96f-outputsessions-0']
```

Comment:
Do these tests depend on files in `gs://leanplum_export`?

Reply:
Nope, no outside dependencies in these tests.